src/unix/async.c

```diff
@@ -63,53 +64,50 @@ int uv_async_init(uv_loop_t* loop, uv_async_t* handle, uv_async_cb async_cb) {
 
 int uv_async_send(uv_async_t* handle) {
   _Atomic int* pending;
-  int expected;
+  _Atomic int* busy;
 
   pending = (_Atomic int*) &handle->pending;
+  busy = (_Atomic int*) &handle->u.fd;
 
   /* Do a cheap read first. */
   if (atomic_load_explicit(pending, memory_order_relaxed) != 0)
     return 0;
 
-  /* Tell the other thread we're busy with the handle. */
-  expected = 0;
-  if (!atomic_compare_exchange_strong(pending, &expected, 1))
-    return 0;
+  /* Set the loop to busy. */
+  atomic_fetch_add(busy, 1);
```

Comment:
Yes, these need to be seqcst, so that the read of the busy flag is sequentially consistently ordered with the write to the pending flag in the spinwait function